telemetry/telemetry.go

```diff
@@ -300,7 +305,19 @@ func WorkflowStep(workflow string, message string) {
 }
 
 // Report command line flags (before any error checking.)
+// This event will be sent before any other events and only once per agent invocation.
 func CommandLine(command string, commandLine []string) {
+	// Look for a service ID in the command line, and assign it to package level variable.
+	for _, arg := range commandLine {
+		if serviceIDRegex.MatchString(arg) {
+			err := akid.ParseIDAs(arg, &serviceID)
+			if err != nil {
+				printer.Warningf("Error parsing service ID from command line: %v\n", err)
+			}
+			break
+		}
```

Comment:
Non-blocking: I really don't see any downside to calling parseIDAs on every argument, instead of pre-filtering with a regular expression?  That is, if we drop the warning.

I am more concerned the other direction, where this warning appears in addition to other error messages explaining the same thing.